src/router.py

```diff
@@ -0,0 +1,159 @@
+from router_leg import RouterLeg    
+import pascy
+import socket
+import select
+from functools import lru_cache
+
+ROUTER_LEGS = []
+
+ARP_TABLE = {'1.1.1.2':'02:42:01:01:01:02',
+             '2.2.2.2':'02:42:02:02:02:02'}
+
+BROADCAST = 'FF:FF:FF:FF:FF:FF'
+
+def get_packet(leg : RouterLeg) -> pascy.Layer:
+    # Recieve packet
+    raw_socket = leg.raw_socket
+    pkt_raw = raw_socket.recvfrom(65536)[0]
+
+    # Parse ethernet layer
+    eth = pascy.l2.EthernetLayer()
+    eth.deserialize(pkt_raw)
+    pkt_raw = pkt_raw[len(eth):]
+
+    dst = pascy.fields.MacAddress.mac2str(eth.dst)
+    # Checks if the packet is for me
+    if dst != leg.mac and dst != BROADCAST:
+        return None
+
+    # Parse arp layer
+    if eth.ether_type == pascy.EthernetLayer.ARP_ETHER_TYPE:
+        arp = pascy.l2.ArpLayer()
+        arp.deserialize(pkt_raw)
+        pkt_raw = pkt_raw[len(arp):]
+        eth.connect_layer(arp)
+
+    # Parse ip layer
+    elif eth.ether_type == pascy.EthernetLayer.IPV4_ETHER_TYPE:
+        ip = pascy.l3.IpLayer()
```

Comment:
Wouldn't it make more sense to split parsing different layers to functions?
Something like:
```python
if eth.ether_type == pascy.EthernetLayer.ARP_ETHER_TYPE:
    parse_arp(eth, pkt_raw)

elif eth.ether_type == pascy.EthernetLayer.IPV4_ETHER_TYPE:
    parse_ip(eth, pkt_raw)
```
This way, `parse_ip` could also handle ICMP, making the code more modular and readable